cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,175 @@
+        cloudformation_dir = os.path.join(current_dir, "..", "..", "..", "..", "cloudformation", "imagebuilder")
+
+        # Components
+        components = []
+        if dev_settings_config and dev_settings_config.update_os_and_reboot:
+            imagebuilder.CfnComponent(
+                self,
+                id="UpdateAndRebootComponent",
+                name="-".join(["UpdateAndReboot", resources_prefix]),
+                version="0.0.1",
+                description="Update OS and Reboot",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(_get_component_data(cloudformation_dir, "update_and_reboot.yaml")),
+            )
+            components.append(
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("UpdateAndRebootComponent")
+                )
+            )
+
+        imagebuilder.CfnComponent(
+            self,
+            id="PClusterComponent",
+            name="-".join(["PCluster", resources_prefix]),
+            version="0.0.1",
+            description="Bake PCluster AMI",
+            change_description="First version",
+            platform="Linux",
+            data=core.Fn.sub(_get_component_data(cloudformation_dir, "pcluster_install.yaml")),
+        )
+
+        components.append(
+            imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(component_arn=core.Fn.ref("PClusterComponent"))
+        )
+
+        # ImageRecipe
+        imagebuilder.CfnImageRecipe(
+            self,
+            id="PClusterImageRecipe",
+            name="-".join(["PCluster", utils.get_installed_version().replace(".", "-"), resources_prefix]),
+            version="0.0.1",
+            parent_image=core.Fn.sub(build_config.parent_image),
+            components=components,
+            block_device_mappings=[
+                imagebuilder.CfnImageRecipe.InstanceBlockDeviceMappingProperty(
+                    device_name="/dev/xvda",
+                    ebs=ebs,
+                )
+            ],
+        )
+
+        # Image
+        imagebuilder.CfnImage(
+            self,
+            id="PClusterImage",
+            image_recipe_arn=core.Fn.ref("PClusterImageRecipe"),
+            infrastructure_configuration_arn=core.Fn.ref("PClusterImageInfrastructureConfiguration"),
+        )
+
+        # AWS Systems Manager
+        ssm.CfnParameter(
+            self,
+            id="PClusterParameter",
+            description="Image Id for PCluster",
+            name="-".join(["/Test/Images/PCluster", resources_prefix]),
+            type="String",
+            value=core.Fn.get_att("PClusterImage", "ImageId").to_string(),
+        )
+
+
+def _get_component_data(source_dir, file_name):
+    """Get component string data from yaml file."""
+    os.path.join(source_dir, file_name)
+    data = load_yaml(os.path.join(source_dir, file_name))
+    return yaml.dump(data)
```

Comment:
let's rename the load_yaml in common to load_yaml_dict.
and you can move this function in common by calling it load_yaml

Reply:
Done